Tests/NIOHTTPCompressionTests/HTTPRequestDecompressorTest.swift

```diff
@@ -63,14 +63,10 @@ class HTTPRequestDecompressorTest: XCTestCase {
         let buffer = ByteBuffer.of(bytes: [120, 156, 75, 76, 28, 5, 200, 0, 0, 248, 66, 103, 17])
         let compressed = compress(buffer, "gzip")
 
-        do {
-            try channel.writeInbound(HTTPServerRequestPart.body(compressed))
-        } catch let error as NIOHTTPDecompression.DecompressionError {
-            switch error {
-            case .limit:
-                // ok
-                break
-            default:
+        XCTAssertThrowsError(try channel.writeInbound(HTTPServerRequestPart.body(compressed))) {error in
+           if case .some(.limit) = error as? NIOHTTPDecompression.DecompressionError {
+                // Okay
+            } else {
```

Comment:
Ok, let's revert this change entirely: as you note, the assertion is not valid here, so we shouldn't bother to make the change.